girder/molecules/molecules/utilities/molecules.py

```diff
@@ -30,22 +30,13 @@ def create_molecule(data_str, input_format, user, public, gen3d=True,
                     provenance='uploaded by user'):
 
     using_2d_format = (input_format in openbabel_2d_formats)
-    smiles_format = 'smiles'
+    inchi_format = 'inchi'
 
     if using_2d_format or input_format in openbabel_3d_formats:
-        smiles = openbabel.to_smiles(data_str, input_format)
+        inchi, inchikey = openbabel.to_inchi(data_str, input_format)
     else:
         sdf_data = avogadro.convert_str(data_str, input_format, 'sdf')
-        smiles = openbabel.to_smiles(sdf_data, 'sdf')
-
-    props = openbabel.properties(smiles, smiles_format)
-    atom_count = props['atomCount']
-
-    if atom_count > 1024:
-        raise RestException('Unable to generate inchi, '
-                            'molecule has more than 1024 atoms .', code=400)
-
-    (inchi, inchikey) = openbabel.to_inchi(smiles, smiles_format)
+        inchi, inchikey = openbabel.to_inchi(sdf_data, 'sdf')
 
     if not inchi:
         raise RestException('Unable to extract inchi', code=400)
```

Comment:
We should capitalize the error properly - InChI

Reply:
Done.